isisd/dict.c

```diff
@@ -1323,6 +1323,7 @@ static void construct(dict_t *d)
 				free(val);
 				if (dn)
 					dnode_destroy(dn);
+				break;
 			}
 
```

Comment:
I think an else would be easier to follow, does this work for you?

Reply:
can you please provide your patch ? I do not catch an idea